src/app/public/modules/toast/toast.component.ts

```diff
@@ -88,14 +103,46 @@ export class SkyToastComponent implements OnInit {
 
   private isOpen = false;
 
+  private autoCloseTimeoutId: any;
+
+  private ngUnsubscribe = new Subject<void>();
+
   private _toastType: SkyToastType;
 
   constructor(
-    private changeDetector: ChangeDetectorRef
+    private changeDetector: ChangeDetectorRef,
+    @Optional() private toasterService?: SkyToasterService
   ) { }
 
   public ngOnInit(): void {
     this.isOpen = true;
+
+    this.startAutoCloseTimer();
+
+    const actionHandler = (value: boolean) => {
+      if (value) {
+        this.stopAutoCloseTimer();
+      } else {
+        this.startAutoCloseTimer();
+      }
+    };
+
+    if (this.toasterService) {
+      this.toasterService.focusIn
+      .takeUntil(this.ngUnsubscribe)
+      .subscribe(actionHandler);
+
+      this.toasterService.mouseOver
+        .takeUntil(this.ngUnsubscribe)
+        .subscribe(actionHandler);
```

Comment:
Unblocking, but we might be able to get away with an observable merge or join for `focusIn` and `mouseOver` since they share the same handler.

Reply:
Updated to use `combineLatest`.